repository/src/main/java/org/apache/atlas/repository/audit/ESBasedAuditRepository.java

```diff
@@ -328,24 +328,25 @@ private boolean createAuditIndex() throws IOException {
         return isSuccess(response);
     }
 
-    private boolean isFieldLimitDifferent() {
-        JsonNode fieldLimit;
+    private boolean shouldUpdateFieldLimitSetting() {
+        JsonNode currentFieldLimit;
         try {
-            fieldLimit = getIndexFieldLimit();
+            currentFieldLimit = getIndexFieldLimit();
         } catch (IOException e) {
             LOG.error("Problem while retrieving the index field limit!", e);
             return false;
         }
         Integer fieldLimitFromConfigurationFile = configuration.getInt(TOTAL_FIELD_LIMIT);
-        return fieldLimit == null || fieldLimitFromConfigurationFile.equals(fieldLimit.intValue());
+        return currentFieldLimit == null || fieldLimitFromConfigurationFile > currentFieldLimit.asInt();
     }
 
     private JsonNode getIndexFieldLimit() throws IOException {
         Request request = new Request("GET", INDEX_NAME + "/_settings");
         Response response = lowLevelClient.performRequest(request);
         ObjectMapper objectMapper = new ObjectMapper();
-        String fieldName = INDEX_NAME + ".settings.index.mapping.total_fields.limit";
-        return objectMapper.readTree(copyToString(response.getEntity().getContent(), Charset.defaultCharset())).get(fieldName);
+        String fieldPath = String.format("/%s/settings/index/mapping/total_fields/limit", INDEX_NAME);
+
+        return objectMapper.readTree(copyToString(response.getEntity().getContent(), Charset.defaultCharset())).at(fieldPath);
```

Comment:
Why are changing this logic?

Reply:
This was never extracting limit value from response, it always returning null, fixed this logic to extract current limit properly

Reply:
ahhh, Cool